MIGRATION.md

```diff
@@ -0,0 +1,130 @@
+
+### Configure
+
+The `configure` widget is now included in the UiState. If you want to exclude this from the URL (because it's usually static, or it allows users to add arbitrary search parameters), you can either use the default stateMappings which exclude it, or exclude it yourself in a custom state mapping. This has to be done both in `stateToRoute` to prevent it appearing in the URL, and `routeToState` to prevent it from applying from the URL. You can check the [source](https://github.com/algolia/instantsearch.js/blob/next/src/lib/stateMappings/simple.ts) for inspiration.
+
+## Helper
+
+This release includes v3 of the `algoliasearch-helper` package. If you are using the built-in widgets, nothing will change for you. This version no longer includes `lodash`, having two main places of impact for users. The main difference will be in bundle size, which is now significantly smaller. If you are using any methods from the `helper` or `searchResults`, please see the detailed changelog [here](https://github.com/algolia/algoliasearch-helper-js/blob/next/documentation-src/metalsmith/content/upgrade.md).
+
+## Custom widgets
+
+As mentioned earlier, any reference of the helper now contains the "new helper", as well as the `getConfiguration` life cycle no longer being used. You can replace its usage with `getWidgetSearchParameters` and `getWidgetState`. Note that this means that your custom widget will also take part in routing from then on. You can still exclude it from the URL via `stateMapping`.
+
+### getConfiguration
+
+This life cycle could be used for setting up a facet, or a default refinement previously. This is now replaced by `getWidgetState` and `getWidgetSearchParameters`. This means that there's no longer a separate step for setting up. However, that can be implemented using a check inside `getWidgetSearchParameters`. Code like this (abbreviated):
+
+```js
+const widget = {
+  getConfiguration(searchParams) {
+    return {
+      disjunctiveFacets: ['myAttribute'],
+    };
+  },
+};
+```
+
+Will become like this:
+
+```js
+const widget = {
+  getWidgetSearchParameters(searchParameters, { uiState }) {
+    return searchParameters
+      .addDisjunctiveFacet('myAttribute')
+      .addDisjunctiveFacetRefinement(
+        'myAttribute',
+        uiState.myWidgetName.myAttribute
+      );
+  },
+  getWidgetState(uiState, { searchParameters }) {
+    return {
+      ...uiState,
+      myWidgetName: {
+        myAttribute: searchParameters.getDisjunctiveRefinements('myAttribute')
+      }
+    };
+  }
+};
+```
+
+While this looks longer, more of the functions are now together than before, and it isn't possible anymore to get into a state where `getConfiguration` is called multiple times and isn't behaving as you would want.
```

Comment:
> more of the functions are now together than before 
could double check this phrase, I don't understand it maybe it's missing a word?

Reply:
Agree the sentence isn't that good, do you know what it should say?